src/sys/optionsreader.cxx

```diff
@@ -74,46 +74,50 @@ void OptionsReader::write(Options *options, const char *file, ...) {
   delete parser;
 }
 
-void OptionsReader::parseCommandLine(Options *options, int argc, char **argv) {
+void OptionsReader::parseCommandLine(Options* options, int argc, char** argv) {
   // A key/value pair, separated by a '=' or a switch
   // and sections separated with an '_' but don't start with a '-'
 
   std::string buffer;
 
   // Go through command-line arguments
-  for (int i=1;i<argc;i++) {
+  for (int i = 1; i < argc; i++) {
 
     // Reset the section
     options = options->getRoot();
 
     buffer = argv[i];
+    if (buffer.length() == 0) {
+      continue;
+    }
     // Test if name starts with a '-', and remove if found
-    if (buffer[0] == '-')
-      buffer = buffer.substr(1);  // Remove the first character (-)
-    
+    if (buffer[0] == '-') {
+      buffer = buffer.substr(1); // Remove the first character (-)
+      if (buffer.length() == 0) {
+        throw BoutException(_("single `-` found - do not know what to do"));
```

Comment:
Maybe "Invalid command line option '-'"? Possibly a reminder to check whitespace too? I guess this is most likely to happen with a typo like `- d`